olive/resource_path.py

```diff
@@ -316,64 +317,126 @@ def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: b
         return str(new_path)
 
 
+def _datastore_url_validator(v, values, **kwargs):
+    aml_info_ready = all([values.get("azureml_client"), values.get("datastore_name"), values.get("relative_path")])
+    if not v and not aml_info_ready:
+        raise ValueError(
+            "If datastore_url is not specified, then azureml_client, datastore_name, and relative_path "
+            "must be specified."
+        )
+    elif v and aml_info_ready:
+        logger.warning("datastore_url is specified. azureml_client, datastore_name, and relative_path are ignored.")
+
+    if v and not v.startswith("azureml:"):
+        raise ValueError(f"Datastore URL {v} is not a valid AzureML datastore URL.")
+    return v
+
+
 class AzureMLDatastore(ResourcePath):
-    """AzureML datastore resource path"""
+    """AzureML DataStore resource path"""
 
     name = ResourceType.AzureMLDatastore
 
+    @staticmethod
+    def _validators() -> Dict[str, Callable[..., Any]]:
+        validators = ResourcePath._validators()
+        validators.update(
+            {
+                "validate_datastore_url": validator("datastore_url", allow_reuse=True)(_datastore_url_validator),
+            }
+        )
+        return validators
+
     @staticmethod
     def _default_config() -> Dict[str, Any]:
         return {
-            "azureml_client": ConfigParam(
-                type_=AzureMLClientConfig, required=True, description="AzureML client config."
-            ),
-            "datastore_name": ConfigParam(type_=str, required=True, description="Name of the datastore."),
-            "relative_path": ConfigParam(type_=str, required=True, description="Relative path to the resource."),
+            "azureml_client": ConfigParam(type_=AzureMLClientConfig, description="AzureML client config."),
+            "datastore_name": ConfigParam(type_=str, description="Name of the datastore."),
+            "relative_path": ConfigParam(type_=str, description="Relative path to the resource."),
+            "datastore_url": ConfigParam(type_=str, description="URL of the datastore."),
         }
 
     def get_path(self) -> str:
-        workspace_config = self.config.azureml_client.get_workspace_config()
-        return (
-            f"{_get_azureml_resource_prefix(workspace_config)}"
-            f"/datastores/{self.config.datastore_name}/paths/{self.config.relative_path}"
-        )
+        if not self.config.datastore_url:
+            workspace_config = self.config.azureml_client.get_workspace_config()
+            return (
+                f"{_get_azureml_resource_prefix(workspace_config)}"
+                f"/datastores/{self.config.datastore_name}/paths/{self.config.relative_path}"
+            )
+        return self.config.datastore_url
+
+    def is_file(self, fsspec) -> bool:
+        return fsspec.info(self.get_relative_path()).get("type") == "file"
+
+    def get_relative_path(self) -> str:
+        if not self.config.relative_path:
```

Comment:
based on the highest priority of `datastore_url`, I think this condition should be `if self.config.datastore_url`?

Reply:
Fixed.